### PR TITLE
Beta v8.14.0

### LINKED ISSUE #6130: Open Beta v8.14 | Please help testing and hardening our upcoming release
RC version | v8.14.2
---------- | -------
v8.14.0 PR | #6078
v8.14.1 PR | #6138
v8.14.2 PR | #6146
Raw changelog | https://github.com/MichaIng/DietPi/blob/beta/CHANGELOG.txt
Code changes | https://github.com/MichaIng/DietPi/compare/master...beta
Release date | **2023-02-11**

#### Related/solved issues: https://github.com/MichaIng/DietPi/issues?q=is%3Aissue+milestone%3Av8.14
____
### Beta v8.14.2
_(2022-02-11)_

#### New images
- Orange Pi 5 | Support for Orange Pi 5 SBC has been added to DietPi with hardware ID 80.
- Odroid N2/HC4 | New images have been generated with support for the default Hardkernel SPI bootloader petitboot. Thanks to this, boot from USB devices is now possible, and on Odroid HC4 it is not necessary anymore to erase the SPI flash or flash mainline U-Boot to it. Many thanks to @hstarcm and @Power-onoff for bringing up this topic and help testing adjustments to the boot script: https://github.com/MichaIng/DietPi/issues/5634
- ROCK 5B | New images with ext4 boot filesystem have been generated, to solve possible issues and limitations of the previously used FAT filesystem. Many thanks to @isarrider for reporting this issue: https://github.com/MichaIng/DietPi/issues/6122

#### New architecture
- Initial experimental support for RISC-V and the StarFive VisionFive 2 SBC has been added. This is all in playground stage currently, but if anyone wants to join the party, contact us.

#### Enhancements
- WiFi | Since Linux 4.15, the kernel can read the WiFi country code from connected access points, and searches the regulatory database for related allowed frequencies automatically. The Central Regulatory Domain Agent (CRDA) has hence become obsolete and isn't even functional if the kernel feature is used. Linux needs to have the regulatory database in specific format stored below `/lib/firmware` for this, which is the case since Debian Bullseye. Additionally, if no Debian kernel is used (all SBCs), one needs to change the used regulatory database from a Debian-only one to the upstream one. New DietPi image ship now with this setup without CRDA, it is applied on next DietPi update on Bullseye and Bookworm systems, and it is offered to purge the obsolete CRDA package. This also fixes the issue on Bookworm SBC systems, where the limited global (00) country code was used, since CRDA is not available on Bookworm, but the Debian-only regulatory database used by default.
- NanoPi R2S/NEO3 | Video related kernel modules are now blacklisted on these headless SBCs, saving some bits of RAM and load at boot.
- DietPi-Globals | G_CHECK_NET: When hitting "Retry" from the error handler, the IPv6 connection check is now skipped as well when IPv6 is disabled via dietpi-config network settings or manually. The same is true for changes to the CONFIG_CHECK_CONNECTION/DOMAIN dietpi.txt settings, which can now also be changed right from the error handler. Many thanks to @derSebastian for reporting a related issue: https://github.com/MichaIng/DietPi/issues/5667
- DietPi-Config | It is now possible to change the dietpi.txt network connection test settings CONFIG_CHECK_CONNECTION_IP, CONFIG_CHECK_CONNECTION_IPV6 and CONFIG_CHECK_DNS_DOMAIN via dietpi-config Network Options: Misc menu. Those are used when checking Internet access via IPv4 and IPv6 and DNS resolver functionality.
- DietPi-Config | The user experience of the network adapter options has been improved:
	- When changing the WiFi country code, the network is not restarted anymore and the check whether the country code has been successfully applied, has been dropped. When the kernel applies allowed frequencies directly (see above WiFi changelog entry), the country code cannot be changed anyway, and there are other cases where this fails. It is not worth the hassle of a full network restart only to check whether the country code is valid.
	- If WiFi settings are applied while no SSID has been configured yet, the interface is not tried to be brought up so that one does not need to wait for the DHCP timeout to continue.
	- The dialogue which offers to purge WiFi packages is now only shown if any of those packages is actually installed.
	- System services are not stopped and restarted anymore when applying network changes, but only hostapd and the DHCP server, if installed.
	- Most WiFi handling is now done with the modern `iw` utility instead of the legacy `wireless-tools` set.
- DietPi-LetsEncrypt | HTTP/2 is now enabled automatically when applying HTTPS on Apache. Nginx and Lighttpd have this enabled OOTB already.
- DietPi-Software | Gogs: Our ARMv6 build for Raspberry Pi 1 and Zero (1) models has been updated to latest version 0.12.10. The update can be applied via reinstall: dietpi-software reinstall 49
- DietPi-Software | Docker Compose: Migrated to the plugin variant, installed from the Docker APT repository. On existing installs, the migration can be done via "dietpi-software reinstall 134". To call it, use `docker compose` (without dash) from now on. Many thanks to @isarrider for making us aware of this: https://github.com/MichaIng/DietPi/issues/6135

#### Bug fixes
- ROCK Pi S | Resolved an issue where the common serial console on UART0 did not show any output and login prompt. Many thanks to @Marsu31 for reporting this issue: https://github.com/MichaIng/DietPi/issues/5972
- ROCK Pi 4 | Resolved an issue where onboard WiFi on ROCK Pi 4C Plus did not work, and potentially other variants than ROCK Pi 4B. Many thanks to `@Marco` for reporting this issue: https://dietpi.com/forum/t/rock4c-no-wifi-hardware-found/15654
- DietPi-Drive_Manager | Resolved an issue where KVM VirtIO virtual disk volumes were not handled correctly. Many thanks to @bennettp123 for reporting this issue: https://github.com/MichaIng/DietPi/issues/6128
- DietPi-Software | Docker: Resolved a v8.13 regression where Docker dependants (Portainer, Roon Extension Manager) failed to install as the Docker socket was not activated. Many thanks to @omavoss for reporting this issue: https://github.com/MichaIng/DietPi/issues/6065
- DietPi-Software | ownCloud/Nextcloud: Resolved an issue where a global software password with dashes lead to a failing install. Many thanks to @maxmoonshark for reporting this issue: https://github.com/MichaIng/DietPi/issues/4199
- DietPi-Software | myMPD: Resolved an issue where intended initial configs were not applied correctly and web UI changes did not survive a service restart.
- DietPi-Software | X.Org X Server: Resolved an issue where the install failed on ARM VMs with KMS support, where the xserver-xorg-video-vmware package was tried to be installed. Many thanks to @mews-se for reporting this issue: https://github.com/MichaIng/DietPi/issues/6085
- DietPi-Software | Amiberry: Resolved an issue where using keyboard LEDs as power/activity indicators did not work, and stopping the service could hang. You can apply the fix by reinstalling Amiberry via "dietpi-software reinstall 108". Many thanks to @DJP75 for reporting this issue: https://github.com/MichaIng/DietPi/issues/5536
- DietPi-Software | Synapse: Resolved an issue where the installation failed on 32-bit ARM systems. Many thanks to `@Dark77` for reporting this issue: https://dietpi.com/forum/t/matrix-synapse-fails-to-install/15831

### PR DESCRIPTION
**Pull request by MichaIng**
### Beta v8.14.0
_(2022-02-04)_

#### New images
- Orange Pi 5 | Support for Orange Pi 5 SBC has been added to DietPi with hardware ID 80.
- Odroid N2/HC4 | New images have been generated with support for the default Hardkernel SPI bootloader petitboot. Thanks to this, boot from USB devices is now possible, and on Odroid HC4 it is not necessary anymore to erase the SPI flash or flash mainline U-Boot to it. Many thanks to @hstarcm and @Power-onoff for bringing up this topic and help testing adjustments to the boot script: https://github.com/MichaIng/DietPi/issues/5634
- ROCK 5B | New images with ext4 boot filesystem have been generated, to solve possible issues and limitations of the previously used FAT filesystem. Many thanks to @isarrider for reporting this issue: https://github.com/MichaIng/DietPi/issues/6122

#### Enhancements
- NanoPi R2S/NEO3 | Video related kernel modules are now blacklisted on these headless SBCs, saving some bits of RAM and load at boot.
- DietPi-Globals | G_CHECK_NET: When hitting "Retry" from the error handler, the IPv6 connection check is now skipped as well when IPv6 is disabled via dietpi-config network settings or manually. The same is true for changes to the CONFIG_CHECK_CONNECTION/DOMAIN dietpi.txt settings, which can now also be changed right from the error handler. Many thanks to @derSebastian for reporting a related issue: https://github.com/MichaIng/DietPi/issues/5667
- DietPi-Config | It is now possible to change the dietpi.txt network connection test settings CONFIG_CHECK_CONNECTION_IP, CONFIG_CHECK_CONNECTION_IPV6 and CONFIG_CHECK_DNS_DOMAIN via dietpi-config Network Options: Misc menu. Those are used when checking Internet access via IPv4 and IPv6 and DNS resolver functionality.
- DietPi-Software | Gogs: Our ARMv6 build for Raspberry Pi 1 and Zero (1) models has been updated to latest version 0.12.10. The update can be applied via reinstall: dietpi-software reinstall 49

#### Bug fixes
- ROCK Pi S | Resolved an issue where the common serial console on UART0 did not show any output and login prompt. Many thanks to @Marsu31 for reporting this issue: https://github.com/MichaIng/DietPi/issues/5972
- ROCK Pi 4 | Resolved an issue where onboard WiFi on ROCK Pi 4C Plus did not work, and potentially other variants than ROCK Pi 4B. Many thanks to `@Marco` for reporting this issue: https://dietpi.com/forum/t/rock4c-no-wifi-hardware-found/15654
- DietPi-Drive_Manager | Resolved an issue where KVM VirtIO virtual disk volumes were not handled correctly. Many thanks to @bennettp123 for reporting this issue: https://github.com/MichaIng/DietPi/issues/6128
- DietPi-Software | Docker: Resolved a v8.13 regression where Docker dependants (Portainer, Roon Extension Manager) failed to install as the Docker socket was not activated. Many thanks to @omavoss for reporting this issue: https://github.com/MichaIng/DietPi/issues/6065
- DietPi-Software | ownCloud/Nextcloud: Resolved an issue where a global software password with dashes lead to a failing install. Many thanks to @maxmoonshark for reporting this issue: https://github.com/MichaIng/DietPi/issues/4199
- DietPi-Software | myMPD: Resolved an issue where intended initial configs were not applied correctly and web UI changes did not survive a service restart.
- DietPi-Software | X.Org X Server: Resolved an issue where the install failed on ARM VMs with KMS support, where the xserver-xorg-video-vmware package was tried to be installed. Many thanks to @mews-se for reporting this issue: https://github.com/MichaIng/DietPi/issues/6085
- DietPi-Software | Amiberry: Resolved an issue where using keyboard LEDs as power/activity indicators did not work, and stopping the service could hang. You can apply the fix by reinstalling Amiberry via "dietpi-software reinstall 108". Many thanks to @DJP75 for reporting this issue: https://github.com/MichaIng/DietPi/issues/5536

### CODE REVIEW

**Comment by Joulinar** [comment]
@MichaIng 
tried to install NC on current `dev` branch on 3 different system. It always failing on same step

```
[FAILED] DietPi-Software | Nextcloud ncc install
 - Command: ncc maintenance:install --no-interaction --database=mysql --database-name=nextcloud --database-user=tmp_root --database-pass=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --admin-user=admin --admin-pass=XXXXXXXX --data-dir=/mnt/dietpi_userdata/nextcloud_data
```

with error `Too many arguments, expected arguments "command".`

**Review by Joulinar — COMMENTED** [review]
Are you sure that this is correct to add `=`? At least following is suggested by the error message

```
maintenance:install [--database DATABASE] [--database-name DATABASE-NAME] [--database-host DATABASE-HOST]         
[--database-port DATABASE-PORT] [--database-user DATABASE-USER] [--database-pass [DATABASE-PASS]]                 
[--database-table-space [DATABASE-TABLE-SPACE]] [--admin-user ADMIN-USER] [--admin-pass ADMIN-PASS] [--admin-email
[ADMIN-EMAIL]] [--data-dir DATA-DIR]              
```

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -4612,7 +4627,7 @@ The install script will now exit. After applying one of the the above, rerun die
 					fi
 
 				}
-				G_EXEC ncc maintenance:install --no-interaction --database 'mysql' --database-name 'nextcloud' --database-user 'tmp_root' --database-pass "${nc_password//?/X}" --admin-user "$username" --admin-pass "${GLOBAL_PW//?/X}" --data-dir "$datadir"
+				G_EXEC ncc maintenance:install --no-interaction --database='mysql' --database-name='nextcloud' --database-user='tmp_root' --database-pass="${nc_password//?/X}" --admin-user="$username" --admin-pass="${GLOBAL_PW//?/X}" --data-dir="$datadir"
```
Comment (Joulinar) [inline dietpi/dietpi-software:4621]: Are you sure that this is correct to add `=`? At least following is suggested by the error message.

```
maintenance:install [--database DATABASE] [--database-name DATABASE-NAME] [--database-host DATABASE-HOST]         
[--database-port DATABASE-PORT] [--database-user DATABASE-USER] [--database-pass [DATABASE-PASS]]                 
[--database-table-space [DATABASE-TABLE-SPACE]] [--admin-user ADMIN-USER] [--admin-pass ADMIN-PASS] [--admin-email
[ADMIN-EMAIL]] [--data-dir DATA-DIR]
```

**EDIT**
Unsing an alternative command without `=` is working well

```
[FAILED] DietPi-Software | Nextcloud ncc install
 - Command: ncc maintenance:install --no-interaction --database=mysql --database-name=nextcloud --database-user=tmp_root --database-pass=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --admin-user=admin --admin-pass=XXXXXXXX --data-dir=/mnt/dietpi_userdata/nextcloud_data
[ INFO ] DietPi-Software | Executing alternative command: ncc maintenance:install --no-interaction --database mysql --database-name nextcloud --database-user tmp_root --database-pass XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --admin-user admin --admin-pass XXXXXXXX --data-dir /mnt/dietpi_userdata/nextcloud_data
Nextcloud was successfully installed
```

Comment (MichaIng) [reply dietpi/dietpi-software:4621]: Can you retry with latest commit? Should now be fixed.

Comment (Joulinar) [reply dietpi/dietpi-software:4621]: Yes way better now 👍

**Comment by Joulinar** [comment]
@MichaIng actually we have an issue with `matrix-synapse` on 32bit systems. Pre-compiled wheels are failing at https://www.piwheels.org/project/matrix-synapse/ . Means we require Rust and Build-Essentials atm. Should I open PR?

**Comment by MichaIng** [comment]
I gave the piwheels maintainer another ping. All these should be solved at once with my PR merged.

**Comment by Joulinar** [comment]
ok I would have a solution ready if needed.

**Comment by MichaIng** [comment]
Let's see on Thursday whether we need it, and if so implement it.

**Pull request merged**